webapp/platform/types/src/apps.ts

```diff
@@ -1,6 +1,8 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
+import {parseISO, isValid} from 'date-fns';
```

Comment:
This package is published separately for use in plugins and such, so I'd rather avoid adding a new dependency to it just for this. I assume I know the answer to this already, but is it possible to do this without adding that extra dependency?

If not, I think we should move these type assertions to a new `utils/post_props.ts` in the `channels` package. I can do that in a followup PR though since it's mostly outside the scope of this PR

Reply:
I change to simply use Date() constructor as validation.